modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -503,9 +550,21 @@ public Assignment getAssignment(User.ID userID, Application.Name applicationName
                 experimentBatch.getLabels().remove(experiment.getLabel());
             }
         }
+        if(LOGGER.isDebugEnabled()) LOGGER.debug("allAssignments: {} ", allAssignments);
```

Comment:
In case where isDebugEnabled() is NOT used then computation of complete message (final String msg: from above code snippet) happens irrespective of the logging LEVEL and that can be avoided by using isDebugEnabled() condition. 

E.g.
In case of "LOGGER.debug("allAssignments: {} ", allAssignments);"  msg will be computed something like this "allAssignments: [{assignment=red, status=EXISTING_ASSIGNMENT}, {assignment=null, status=NEW_ASSIGNMENT}]" even if logging level is not DEBUG. I am trying to avoid this String message computation when not needed.